File: Molinos.Scato.WebPuerto/src/app/shared/modelos/embarque-Informacion.ts
import { NumericLiteral } from 'typescript';
import { DateNFOption } from 'xlsx';
import { Bandera } from './bandera';

export class EmbarqueInformacion {
            id :number;
           embarqueId: number;
            imo:string;
            mmsi :string;
            bandera : Bandera;
            tonelaje :number;
            tonelajePesoMuerto:number; 
            largoxAnchoExtremo :string;
            fotoEmbarque:string;
            fechaRegistro:DateNFOption;


            constructor(id,embarque_id,imo,mmsi,bandera,tonelaje,tonelajePesoMuerto,largoxAnchoExtremo,fotoembarque,fecharegistro){
                this.id = id;
                this.embarqueId = embarque_id;
                this.imo = imo;
                this.mmsi = mmsi;
                this.bandera = bandera;
                this.tonelaje = tonelaje;
                this.tonelajePesoMuerto = tonelajePesoMuerto;
                this.largoxAnchoExtremo = largoxAnchoExtremo;
                this.fotoEmbarque = fotoembarque;
                this.fechaRegistro = fecharegistro;
            }
}

File: Molinos.Scato.Dominio/Entidades/EmbarqueInformacion.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Molinos.Scato.Dominio.Entidades
{
    [Table("EmbarqueInformacion")]
    public class EmbarqueInformacion : IIdentificable
    {
  
        [Key, Column(Order = 0)]
        public virtual int Id { get; set; }
        //   [Required]
        public virtual Embarque Embarque { get; set; }
        public virtual  string IMO { get; set; }
        public virtual  string MMSI { get; set; }
        public virtual Bandera Bandera { get; set; }
        public virtual  int Tonelaje { get; set; }
        public virtual  int TonelajePesoMuerto { get; set; }
        public virtual  string LargoxAnchoExtremo { get; set; }
        public virtual  string FotoEmbarque { get; set; }
        public virtual DateTime FechaRegistro { get; set; }


    }
}
